tests/test_bootstrap.py

```diff
@@ -0,0 +1,556 @@
+            mock_glob.reset_mock()
+            mock_shprint.reset_mock()
+            mock_build_dir.reset_mock()
+            mock_bs_dir.reset_mock()
+
+        # test distribute_libs
+        mock_glob.return_value = [
+            "/fake_dir/libsqlite3.so",
+            "/fake_dir/libpng16.so",
+        ]
+        bs.distribute_libs(arch, [self.ctx.get_libs_dir(arch.arch)])
+        libs_dir = os.path.join("libs", arch.arch)
+        # we expect two calls to glob/copy command via shprint
+        self.assertEqual(len(mock_glob.call_args_list), 2)
+        self.assertEqual(len(mock_shprint.call_args_list), 2)
+        for i, lib in enumerate(mock_glob.return_value):
+            self.assertEqual(
+                mock_shprint.call_args_list[i],
+                mock.call(sh.cp, "-a", lib, libs_dir),
+            )
+        mock_build_dir.assert_called()
+        mock_bs_dir.assert_called_once_with(libs_dir)
+        reset_mocks()
+
+        # test distribute_javaclasses
+        mock_glob.return_value = ["/fakedir/java_file.java"]
+        bs.distribute_javaclasses(self.ctx.javaclass_dir)
+        mock_glob.assert_called_once_with(self.ctx.javaclass_dir)
+        mock_build_dir.assert_called_with(self.ctx.javaclass_dir)
+        mock_bs_dir.assert_called_once_with("src")
+        self.assertEqual(
+            mock_shprint.call_args,
+            mock.call(sh.cp, "-a", "/fakedir/java_file.java", "src"),
+        )
+        reset_mocks()
+
+        # test distribute_aars
+        mock_glob.return_value = ["/fakedir/file.aar"]
+        bs.distribute_aars(arch)
+        mock_build_dir.assert_called_with(self.ctx.aars_dir)
+        # We expect three calls to shprint: unzip, cp, cp
+        zip_call, kw = mock_shprint.call_args_list[0]
+        self.assertEqual(zip_call[0], sh.unzip)
+        self.assertEqual(zip_call[2], "/fakedir/file.aar")
+        cp_java_call, kw = mock_shprint.call_args_list[1]
+        self.assertEqual(cp_java_call[0], sh.cp)
+        self.assertTrue(cp_java_call[2].endswith("classes.jar"))
+        self.assertEqual(cp_java_call[3], "libs/file.jar")
+        cp_libs_call, kw = mock_shprint.call_args_list[2]
+        self.assertEqual(cp_libs_call[0], sh.cp)
+        self.assertEqual(cp_libs_call[2], "/fakedir/file.aar")
+        self.assertEqual(cp_libs_call[3], libs_dir)
+        mock_bs_dir.assert_has_calls([mock.call("libs"), mock.call(libs_dir)])
+        mock_glob.assert_called()
+        reset_mocks()
+
+    @mock.patch("pythonforandroid.bootstrap.shprint")
+    @mock.patch("pythonforandroid.bootstrap.sh.Command")
+    @mock.patch("pythonforandroid.build.ensure_dir")
+    @mock.patch("pythonforandroid.archs.find_executable")
+    def test_bootstrap_strip(
+        self,
+        mock_find_executable,
+        mock_ensure_dir,
+        mock_sh_command,
+        mock_sh_print,
+    ):
+        mock_find_executable.return_value = "arm-linux-androideabi-gcc"
+        mock_sh_command.return_value = mock.Mock()
```

Comment:
Minor: this is (already) a `MagicMock` by default, why changing it to a `Mock`? Let's be lazy unless required not to